### PR TITLE
CosmosDb: Implement SaveChanges functionality

### PR DESCRIPTION
**Pull request by smitpatel**
Part of #8443

### CODE REVIEW

File: src/EFCore.Cosmos.Sql/Storage/Internal/CosmosSqlDatabase.cs [hunk 1]
```diff
@@ -3,30 +3,64 @@
...
+
+                Debug.Assert(!entityType.IsAbstract());
+
+                if (!_documentCollections.TryGetValue(entityType, out var documentCollection))
+                {
+                    _documentCollections.Add(
+                        entityType, documentCollection = _documentCollectionFactory.Create(entityType));
+                }
+
+                await documentCollection.SaveAsync(entry);
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos.Sql/Storage/Internal/CosmosSqlDatabase.cs]: no cancellationToken ?

File: src/EFCore.Cosmos.Sql/Storage/Internal/DocumentCollection.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+{
+    public class DocumentCollection<TKey> : IDocumentCollection
+    {
+        private DocumentClient _documentClient;
+        private string _databaseId;
+        private string _collectionName;
+        private readonly IEntityType _entityType;
+        private IPrincipalKeyValueFactory<TKey> _principalKeyValueFactory;
+
+
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos.Sql/Storage/Internal/DocumentCollection.cs]: extra line

File: src/EFCore.Cosmos.Sql/Storage/Internal/DocumentCollection.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            foreach (var property in _entityType.GetProperties())
+            {
+                document.SetPropertyValue(property.Name, entry.GetCurrentValue(property));
+            }
+
+            // Set Discriminator Property for updates
+            document.SetPropertyValue(
+                _entityType.CosmosSql().DiscriminatorProperty.Name,
+                _entityType.CosmosSql().DiscriminatorValue);
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos.Sql/Storage/Internal/DocumentCollection.cs]: Consider only doing this if the discriminator is not set.

File: src/EFCore.Cosmos.Sql/Storage/Internal/DocumentCollection.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos.Sql/Storage/Internal/DocumentCollection.cs]: extra line

**Comment by smitpatel** [comment]
Merged via cd9e9349e9cbd6b0e6326502a0b11a80cc39afd7

**Pull request closed without merging**